frontend/public/components/utils/firehose.jsx

```diff
@@ -116,8 +116,9 @@ const ConnectToState = connect(({k8s}, {reduxes}) => {
   {inject(props.children, _.omit(props, ['children', 'className', 'reduxes']))}
 </div>);
 
-const stateToProps = ({k8s}, {resources}) => ({
+const stateToProps = ({k8s}, {resources, reduxIDs = []}) => ({
   k8sModels: resources.reduce((models, {kind}) => models.set(kind, k8s.getIn(['RESOURCES', 'models', kind])), ImmutableMap()),
+  loaded: reduxIDs.every(id => k8s.getIn([id, 'loaded'])),
```

Comment:
@spadgett So this solution prevents the entire view from being re-rendered, but _something_ is causing `Firehose` to be unmounted when `RESOURCES.inFlight` changes, which causes the WebSockets to be briefly terminated. However, the re-rendering only affects the list rows now and is pretty fast.